### PR TITLE
when readtimer interval end, you can Again.

### PR DESCRIPTION
**Pull request by greatyingzi**
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9844)

### CODE REVIEW

File: plugins/readtimer.koplugin/main.lua [hunk 1]
```diff
@@ -18,9 +22,45 @@ function ReadTimer:init()
...
+                        {
+                            text = _("Cancel"),
+                            id = "close",
+                            callback = function()
+                                self.last_interval_time = 0
+                                UIManager:close(self.dialog)
+                            end,
+                        },
+                        {
+                            -- @translators Again
```
Comment (Frenzie) [inline plugins/readtimer.koplugin/main.lua:43]:
```suggestion
```

File: plugins/readtimer.koplugin/main.lua [hunk 2]
```diff
@@ -18,9 +22,45 @@ function ReadTimer:init()
         if self.time == 0 then return end
 
         self.time = 0
-        UIManager:show(InfoMessage:new{
-            text = T(_("Read timer alarm\nTime's up. It's %1 now."), os.date("%c")),
-        })
+        if self.timer_way == "interval_time" then
+            logger.dbg("timer_way is interval_time, show dialog")
+            self.dialog = ButtonDialogTitle:new{
```
Comment (hius07) [inline plugins/readtimer.koplugin/main.lua:27]: It is simplier to have ConfirmBox here.

File: plugins/readtimer.koplugin/main.lua [hunk 3]
```diff
@@ -18,9 +22,45 @@ function ReadTimer:init()
...
+                title = T(_("Read timer alarm\nTime's up. It's %1 now.", os.date("%c"))),
+                title_align = "center",
+                width_factor = 0.8,
+                use_info_style = false,
+                buttons = {
+                    {
+                        {
+                            text = _("Cancel"),
+                            id = "close",
```
Comment (hius07) [inline plugins/readtimer.koplugin/main.lua:36]: Not needed (there is no input field).

File: plugins/readtimer.koplugin/main.lua (hunk 2 above)
Comment (hius07) [reply plugins/readtimer.koplugin/main.lua:27]: And can be local.

File: plugins/readtimer.koplugin/main.lua (hunk 1 above)
Comment (greatyingzi) [reply plugins/readtimer.koplugin/main.lua]: done

File: plugins/readtimer.koplugin/main.lua (hunk 3 above)
Comment (greatyingzi) [reply plugins/readtimer.koplugin/main.lua]: done

File: plugins/readtimer.koplugin/main.lua [hunk 4]
```diff
@@ -10,6 +11,9 @@ local T = require("ffi/util").template
 local ReadTimer = WidgetContainer:extend{
     name = "readtimer",
     time = 0,  -- The expected time of alarm if enabled, or 0.
+    dialog = nil,
```
Comment (hius07) [inline plugins/readtimer.koplugin/main.lua:14]: leftover

File: plugins/readtimer.koplugin/main.lua [hunk 5]
```diff
@@ -10,6 +11,9 @@ local T = require("ffi/util").template
...
+    last_interval_time = 0,
+    timer_way = nil
```
Comment (hius07) [inline plugins/readtimer.koplugin/main.lua:16]: It can be just true/false, no need of strings.

File: plugins/readtimer.koplugin/main.lua [hunk 6]
```diff
@@ -18,9 +22,26 @@ function ReadTimer:init()
         if self.time == 0 then return end
 
         self.time = 0
-        UIManager:show(InfoMessage:new{
-            text = T(_("Read timer alarm\nTime's up. It's %1 now."), os.date("%c")),
-        })
+        local tip_text = T(_("Read timer alarm\nTime's up. It's %1 now.", os.date("%c")))
+        local confirb_box
```
Comment (hius07) [inline plugins/readtimer.koplugin/main.lua:26]: "confirm"
Better to move inside "if".

Comment (greatyingzi) [reply plugins/readtimer.koplugin/main.lua]: done

File: plugins/readtimer.koplugin/main.lua (hunk 5 above)
Comment (greatyingzi) [reply plugins/readtimer.koplugin/main.lua]: done

File: plugins/readtimer.koplugin/main.lua (hunk 4 above)
Comment (greatyingzi) [reply plugins/readtimer.koplugin/main.lua]: done

File: plugins/readtimer.koplugin/main.lua [hunk 7]
```diff
@@ -125,6 +146,7 @@ function ReadTimer:addToMainMenu(menu_items)
                 text = _("Set interval"),
                 keep_menu_open = true,
                 callback = function(touchmenu_instance)
+                    self.can_repeat = true
```
[the remark this answers is not in the document]
Comment (greatyingzi) [reply plugins/readtimer.koplugin/main.lua]: done

**Comment by hius07** [comment]
Look, `can_repeat = (last_interval_time ~= 0)`, no need to have two triggers.

**Comment by greatyingzi** [comment]
> Look, `can_repeat = (last_interval_time ~= 0)`, no need to have two triggers.

done

**Comment by greatyingzi** [comment]
> Look, `can_repeat = (last_interval_time ~= 0)`, no need to have two triggers.
@hius07 
sorry, I am wrong to close. please reopen this MR

**Comment by greatyingzi** [comment]
> Look, `can_repeat = (last_interval_time ~= 0)`, no need to have two triggers.

**Comment by hius07** [comment]
Thanks, looks good.
Asking @offset-torque for the best button wording: "Again", "Set timer", "Repeat" or smth else.

**Comment by offset-torque** [comment]
There is no picture here but from the code I think plugin says **"Time is up"** and shows two buttons.
- I think **Repeat** should be the clearest, if the function of the button is **"restart the last interval I set"**.
- Other button can be **Done** if its function is **"ending the timer operation"**.

And thanks for the improvement **greatyingzi**.

**Comment by greatyingzi** [comment]
> There is no picture here but from the code I think plugin says **"Time is up"** and shows two buttons.
> 
> * I think **Repeat** should be the clearest, if the function of the button is **"restart the last interval I set"**.
> * Other button can be **Done** if its function is **"ending the timer operation"**.
> 
> And thanks for the improvement **greatyingzi**.

Done

**Pull request merged**